File: examples/stream/listChannels/go/listChannels.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/stream"
)

// List all the active channels
func ListChannels() {
	streamService := stream.NewStreamService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := streamService.ListChannels(&stream.ListChannelsRequest{})
	fmt.Println(rsp, err)
}

File: examples/stream/listChannels/node/listChannels.js
const { StreamService } = require("micro-js-client/stream");

// List all the active channels
async function listChannels() {
  let streamService = new StreamService(process.env.MICRO_API_TOKEN);
  let rsp = await streamService.listChannels({});
  console.log(rsp);
}

listChannels();
